yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -258,7 +258,8 @@ private[spark] class Client(
     if (executorMem > maxMem) {
       throw new IllegalArgumentException(s"Required executor memory (${args.executorMemory}" +
         s"+$executorMemoryOverhead MB) is above the max threshold ($maxMem MB) of this cluster! " +
-        "Please increase the value of 'yarn.scheduler.maximum-allocation-mb'.")
+        "Please increase the value of 'yarn.scheduler.maximum-allocation-mb' and" +
```

Comment:
I've switched it to and/or + check the values. I think its a useful error for people trying to run Spark on YARN who might not know what the configuration params are they need here (much better than the alternative of dumping a stack trace to nowhere in particular).